pysmt/solvers/btor.py

```diff
@@ -275,13 +279,16 @@ def get_named_unsat_core(self):
         if self.options.unsat_cores_mode == "named":
             unsat_core = {}
             # relies on this assertion stack being ordered
+            lst = list(self._named_assertions.keys()) + self._last_assumptions
             assert isinstance(self._assertion_stack, list)
-            btor_named_assertions = [self.converter.convert(a) for a in self._named_assertions.keys()]
+            btor_named_assertions = [self.converter.convert(a) for a in lst]
             in_unsat_core = self.btor.Failed(*btor_named_assertions)
-            for a, in_core in zip(self._assertion_stack, in_unsat_core):
+            cnt = 0
+            for a, in_core in zip(lst, in_unsat_core):
                 if in_core:
-                    name = self._named_assertions[a]
+                    name = self._named_assertions.get(a, '_a_%d' % cnt)
```

Comment:
We are making up the name for the assumption? If so, how is the caller supposed to link this information back? I wonder if in the named case returning the assumptions makes sense.